### PR TITLE
[breaking][deps] babel 7, and other deps

### PR DESCRIPTION
**Pull request by noratarano**
### Summary

Update to babel 7, and updates other deps in preparation for the react-with-styles update. I fought CI for a while, but it should be all good now.

Breaking changes:

1. Drops support for node < 6

### Reviewers

@ljharb @ahuth @TaeKimJR @joeuy @indiesquidge @lencioni

### CODE REVIEW

**Comment by noratarano** [comment]
I'm hoping these updates will fix some of the CI issues from https://github.com/airbnb/rheostat/pull/253

File: package.json [hunk 1]
```diff
@@ -57,20 +57,22 @@
     "eslint": "^6.1.0",
     "eslint-config-airbnb": "^18.0.1",
     "eslint-plugin-import": "^2.18.2",
+    "eslint-plugin-jest": "^22.16.0",
```
Comment (noratarano) [inline package.json]: I'm not using this directly... not sure how to configure it. cc @ljharb @ahuth WDYT?

Comment (ljharb) [reply package.json]: fine if you want to use it; there's no public airbnb for doing so tho

Comment (ahuth) [reply package.json]: I'm cool with it if we like the rules this gives us. I'm not familiar with what they are, though

File: test/.eslintrc [hunk 2]
```diff
@@ -1,7 +1,10 @@
 {
   "env": {
-    "mocha": true
+    "jest": true
```
Comment (ahuth) [inline test/.eslintrc]: 🤘

File: package.json [hunk 3]
```diff
@@ -57,43 +59,49 @@
...
-    "react-with-styles-interface-css": "^4.0.3"
+    "react-with-styles-interface-css": "^5.0.0"
   },
   "jest": {
     "testMatch": [
       "<rootDir>/test/*-test.*"
     ],
     "setupFiles": [
-      "<rootDir>/test/_setup.js"
+      "<rootDir>/test/_helpers/setupFiles.js"
```
Comment (TaeKimJR) [inline package.json]: 👍 Nice rename for clarity

**Pull request merged**